File: myExpenses/src/forTest/java/org/totschnig/myexpenses/provider/PayeeInfo.java
package org.totschnig.myexpenses.provider;

import android.content.ContentValues;

import org.totschnig.myexpenses.util.Utils;

import static org.totschnig.myexpenses.provider.DatabaseConstants.KEY_PAYEE_NAME;
import static org.totschnig.myexpenses.provider.DatabaseConstants.KEY_PAYEE_NAME_NORMALIZED;

public class PayeeInfo {
  public PayeeInfo(String name) {
    this.name = name;
  }

  String name;

  public ContentValues getContentValues() {
    ContentValues contentValues = new ContentValues(2);
    contentValues.put(KEY_PAYEE_NAME, name);
    contentValues.put(KEY_PAYEE_NAME_NORMALIZED, Utils.normalize(name));
    return contentValues;
  }
}

File: myExpenses/src/androidTest/java/org/totschnig/myexpenses/test/provider/PayeeInfo.kt
package org.totschnig.myexpenses.test.provider

import android.content.ContentValues
import org.totschnig.myexpenses.provider.DatabaseConstants.KEY_PARENTID
import org.totschnig.myexpenses.provider.DatabaseConstants.KEY_PAYEE_NAME
import org.totschnig.myexpenses.provider.DatabaseConstants.KEY_PAYEE_NAME_NORMALIZED
import org.totschnig.myexpenses.util.Utils


data class PayeeInfo @JvmOverloads constructor(val name: String, val parentId: Long? = null) {
    val contentValues: ContentValues
        get() = ContentValues(2).apply {
            put(KEY_PAYEE_NAME, name)
            put(KEY_PAYEE_NAME_NORMALIZED, Utils.normalize(name))
            put(KEY_PARENTID, parentId)
        }
}